### PR TITLE
Create job schedule

### PR DESCRIPTION
**Pull request by jweill-aws**
Partial fix for #80.

Adds UI elements to choose between "run now" and "run on a schedule." The API has been implemented, but the job definition list view has not (see #93).

This change also makes the environment picker's label be inside the UI control instead of just above it. This required a small change to the theme.

Demo:

https://user-images.githubusercontent.com/93281816/193711578-d23c7918-77e7-45b6-b777-b12cbb605670.mov

### CODE REVIEW

**Comment by 3coins** [comment]
Looks great!

File: src/components/create-schedule-options.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        name={props.name}
+        value={props.createType}
+        onChange={props.handleCreateTypeChange}
+      >
+        <FormControlLabel
+          value="Job"
+          control={<Radio />}
+          label={trans.__('Run now')}
+        />
+        <Typography sx={{ fontSize: '90%', marginLeft: '2.5em' }}>
```
Comment (ellisonbg) [inline src/components/create-schedule-options.tsx:51]: It is important for us to pull all typography from the JupyterLab theme (size, line height, color). In an MUI context, we could either use a plain HTML element and style it using the JupyterLab theme variables, or we could put it inside the `Typography` component and edit the theme to have MUI use the JupyterLab font style information. My only concern about that is that MUI may be using the typography styling in other contexts that we don't want to restyle.

File: src/components/create-schedule-options.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          {trans.__('Run your notebook immediately')}
```
Comment (ellisonbg) [inline src/components/create-schedule-options.tsx:52]: I don't think the subtitles are adding much here and add visual clutter. Can we just use the text after the radio button to communicate this to the user?

File: src/components/create-schedule-options.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        </Typography>
+        <FormControlLabel
+          value="JobDefinition"
+          control={<Radio />}
+          label={trans.__('Run on a schedule')}
+        />
+        <Typography sx={{ fontSize: '90%', marginLeft: '2.5em' }}>
+          {trans.__('Schedule a notebook at a regular interval')}
```
Comment (ellisonbg) [inline src/components/create-schedule-options.tsx:60]: Ditto here. Also, let's be careful in our language to not assume that the jobs are notebooks. The model here allows an environment to run file as a job.

File: src/components/schedule-inputs.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  const trans = useTranslator('jupyterlab');
+
+  const timezones = Object.keys(tzdata.zones).sort();
+
+  const timezoneLabel = trans.__('Time zone');
+
+  return (
+    <>
+      <TextField
+        label={trans.__('Cron expression')}
```
Comment (ellisonbg) [inline src/components/schedule-inputs.tsx:26]: Is there a regular expression we can use the validate the CRON expression?

**Review by ellisonbg — CHANGES_REQUESTED** [review]

File: src/components/create-schedule-options.tsx (hunk 2 above)
Comment (jweill-aws) [reply src/components/create-schedule-options.tsx:52]: I added these for parity with the Figma mocks, which use them below each label.

File: src/components/schedule-inputs.tsx (hunk 4 above)
Comment (jweill-aws) [reply src/components/schedule-inputs.tsx:26]: There are several, which are hideously long: https://www.codeproject.com/Tips/5299523/Regex-for-cron-expressions

We could apply some simple regexes, like checking for non-blank values or checking for four spaces between the expression's components, as an alternative.

Comment (jweill-aws) [reply src/components/schedule-inputs.tsx:26]: [cron-validate](https://www.npmjs.com/package/cron-validate) can do it all for me.

File: package.json [hunk 5]
```diff
@@ -71,9 +71,11 @@
     "@mui/material": "^5.10.6",
     "@mui/system": "^5.10.6",
     "@types/react-dom": "^18.0.5",
+    "cron-validate": "^1.4.3",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
-    "react-select": "^5.4.0"
+    "react-select": "^5.4.0",
```
Comment (andrii-i) [inline package.json:77]: Is react-select used anywhere? Only see Select being imported from mui in the code

Comment (jweill-aws) [reply package.json:77]: I previously used it for the environment selector; it shouldn't be needed anymore now that we use MUI instead.

**Review by andrii-i — APPROVED** [review]
Looks good to me. 
Downloaded the branch. Everything builds and works correctly

**Pull request merged**